pkg/config/application_test.go

```diff
@@ -516,3 +516,66 @@ func TestGenericPostSyncConfiguration(t *testing.T) {
 		})
 	}
 }
+
+func TestCustomSyncConfig(t *testing.T) {
+	testcases := []struct {
+		fileName           string
+		expectedKind       Kind
+		expectedAPIVersion string
+		expectedSpec       interface{}
+		expectedError      error
+	}{
+		{
```

Comment:
This stage should require `runs` as it's configuration since defining this stage without runs make no sense. You can refer to other stage validation to know how to add validate step for config.